arch/x86/x86.c

```diff
@@ -45,6 +45,20 @@ static void cpuid(int info, unsigned* eax, unsigned* ebx, unsigned* ecx, unsigne
 #endif
 }
 
+static void cpuidex(int info, int subinfo, unsigned* eax, unsigned* ebx, unsigned* ecx, unsigned* edx) {
+#ifdef _MSC_VER
+    unsigned int registers[4];
+    __cpuidex(registers, info, subinfo);
+
+    *eax = registers[0];
+    *ebx = registers[1];
+    *ecx = registers[2];
+    *edx = registers[3];
+#else
+    __get_cpuid_count(info, subinfo, eax, ebx, ecx, edx);
```

Comment:
Perhaps we should use `__get_cpuid` in cpuid() too. What do you think?

Reply:
__get_cpuid doesn't clear ecx register either... Only thing it does extra is check the maximum leaf number.

Reply:
I was suggesting we standardize and use __get_cpuid in cpuid() and __get_cpuid_count in cpuidex(). Otherwise we should stick to __cpuid in cpuid() and __cpuid_count in cpuidex().

Reply:
Seems that __get_cpuid and __get_cpuid_count cause some tests to fail... I can't open the logs to see why they fail, but __cpuid and __cpuid_count don't cause test fails, so I see no reason why we can't use them as we already test for maximum leaf number.

Reply:
Sounds good.